src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -95,6 +95,20 @@ class LinalgBackendBase
 	#undef BACKEND_GENERIC_DOT
 
 	/**
+	* Wrapper method that computes mean of SGVectors and SGMatrices.
+	*
+	* @see linalg::mean
+	*/
+	#define BACKEND_GENERIC_MEAN(Type, Container) \
+	virtual Type mean(const Container<Type>& a) const \
```

Comment:
yeah but why not just:
`virtual float64_t mean(const Container<Type>& a) const`?